src/main/java/frc/trigon/robot/subsystems/elevator/simulationelevator/SimulationElevatorIO.java

```diff
@@ -0,0 +1,48 @@
+package frc.trigon.robot.subsystems.elevator.simulationelevator;
+
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import frc.trigon.robot.simulation.ElevatorSimulation;
+import frc.trigon.robot.subsystems.elevator.ElevatorConstants;
+import frc.trigon.robot.subsystems.elevator.ElevatorIO;
+import frc.trigon.robot.subsystems.elevator.ElevatorInputsAutoLogged;
+import frc.trigon.robot.utilities.Conversions;
+
+public class SimulationElevatorIO extends ElevatorIO {
+    private final ElevatorSimulation motor = SimulationElevatorConstants.MOTOR;
+    private final MotionMagicVoltage positionRequest = new MotionMagicVoltage(0);
+
+    @Override
+    protected void updateInputs(ElevatorInputsAutoLogged inputs) {
+        inputs.motorVoltage = motor.getVoltage();
+        inputs.positionMeters = getMotorPositionMeters();
+        inputs.velocityMetersPerSecond = getMotorVelocityMetersPerSecond();
+        inputs.profiledSetpointMeters = getProfiledSetpointMeters();
+    }
+
+    @Override
+    protected void setTargetPosition(double targetPositionMeters) {
+        motor.setControl(positionRequest.withPosition(SimulationElevatorConstants.PROFILED_PID_CONTROLLER.getGoal().position));
+
+    }
```

Comment:
1. extra newline
2. you don't need a pid controller, the idea is that the simulation has one built in. You just need to input the target position (in revolutions, not meters) and the sim will do the rest.